multicluster/service-mirror/cluster_watcher.go

```diff
@@ -984,21 +985,28 @@ func (rcsw *RemoteClusterServiceWatcher) resolveGatewayAddress() ([]corev1.Endpo
 	var gatewayEndpoints []corev1.EndpointAddress
 	var errors []error
 	for _, addr := range strings.Split(rcsw.link.GatewayAddress, ",") {
-		ipAddr, err := net.ResolveIPAddr("ip", addr)
-		if err == nil {
-			gatewayEndpoints = append(gatewayEndpoints, corev1.EndpointAddress{
-				IP: ipAddr.String(),
-			})
-		} else {
+		ipAddrs, err := net.LookupIP(addr)
+		if err != nil {
 			err = fmt.Errorf("Error resolving '%s': %w", addr, err)
 			rcsw.log.Warn(err)
 			errors = append(errors, err)
+			continue
+		}
+
+		for _, ipAddr := range ipAddrs {
+			gatewayEndpoints = append(gatewayEndpoints, corev1.EndpointAddress{
+				IP: ipAddr.String(),
+			})
 		}
 	}
 	// one resolved address is enough
 	if len(gatewayEndpoints) > 0 {
 		return gatewayEndpoints, nil
 	}
+
+	sort.SliceStable(gatewayEndpoints, func(i, j int) bool {
+		return gatewayEndpoints[i].IP < gatewayEndpoints[j].IP
+	})
```

Comment:
I made it to an early return in case of error so it should actually sort IPs now, I can remove the sort if you think that's not necessary though :+1: